### ISSUE TITLE
hide Brave Rewards icon in address bar

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#1475]
## Description 

I've had several users ask if there was a way for them to disable or hide the Brave Rewards icon in the URL bar: 
![image](https://user-images.githubusercontent.com/3299098/46711233-484ef300-cc00-11e8-8aee-e02f3b5d74ea.png)

Users should be able to disable or hide the icon if they're not interested in the program. Very minor/cosmetic but worth considering.

## Steps to Reproduce

N/A 
## Brave version (chrome://version info)

Dev/Beta builds, v0.55.11 and up

### Reproducible on current release:
N/A

### Website problems only:  
N/a
### Additional Information
Two reports with a few users requesting this in each thread, have a couple Reddit posts to add as well then I find them: 
https://community.brave.com/t/q-remove-brave-rewards/34515/7
(ignore thread title, request is in comments) https://community.brave.com/t/search-bar-by-address-bar/34388/14

**Comment by srirambv** [comment]
+1 on disabling BAT logo when user dismisses the first notification and until rewards is enabled.

**Comment by xv** [comment]
+1
I didn't uninstall Chrome only to find myself ending up with more crap I don't want.

**Comment by sobolevn** [comment]
There one more thing that bothers me.
This red intention-looking notification that I can not switch off or accept.

<img width="83" alt="2018-12-06 10 54 43" src="https://user-images.githubusercontent.com/4660275/49569606-740f0080-f945-11e8-8509-c1cffed3157a.png">

It would be awesome to switch these two icons off. Especially Rewards.

**Comment by snertev** [comment]
+1 to **disable** the whole extension as long as the user doesn't join the program.

It should be a default for each Brave extension that is not used.

**Comment by volcbs** [comment]
+1 too
I want to use Brave, but I really can't ignore the wallet icon that I don't want to use it at least right now.

**Comment by davidtemkin** [comment]
@sobolevn If you click to view the alert, once, the red dot will go away. It disappears as soon as you view the popup, regardless of whether your enable Brave Rewards.

**Comment by sobolevn** [comment]
@davidtemkin thanks for the suggestion! But, sadly it is not the case for me. That's a bug I guess.

**Comment by davidtemkin** [comment]
That is indeed a bug! Thank you @sobolevn , we will address.

**Comment by anonrig** [comment]
The main reason why I switched from Chrome to Brave is to get rid of shitty things like this triangle.

**Comment by eljuno** [comment]
+1 from Twitter https://twitter.com/Bakhita_now/status/1072949012205060096?s=09

**Comment by SylentBobNJ** [comment]
+1 please remove this unneeded icon

**Comment by Rundik** [comment]
+1

**Comment by Qix-** [comment]
Stop saying +1, it doesn't get it finished any faster. Use the emoji responses so everyone and their grandmother don't get notifications for it...

**Comment by harrypnyce** [comment]
> Stop saying +1

Bigger, better, faster, more... _Git`er done, fellas!~_  Can't stop, won't stop.

**Comment by pedrogarcia** [comment]
Why do I have this stupid "Brave Rewards" icon in my browser? 

It's like an horrible ad, that stays turned on and show a horrible notification in the address bar, and is tracking all the pages I visit, with no way to completely disable this horrible feature. 

I do not want it, and I wish to remove it!

**Comment by maikeldotuk** [comment]
Any news on this? Is it going to be removed or should we remove brave browser altogether?

**Comment by maikeldotuk** [comment]
So how do we disable it ?On 30 Dec 2018 18:28, Ivan Alexandrov <notifications@github.com> wrote:I did it already

**Comment by gasman0015** [comment]
I will not use Brave until this obnoxious problem is fixed.

**Comment by maikeldotuk** [comment]
It feels to me as if this is the whole reason for the existence of this browser. Anonymity is the freebie, convincing you of using the rewards is the real agenda On 1 Jan 2019 14:07, gasman0015 <notifications@github.com> wrote:I will not use Brave until this obnoxious problem is fixed.

**Comment by Qix-** [comment]
> Anonymity is the freebie, convincing you of using the rewards is the real agenda

Then what's the point of switching to brave? I'm not joining a cause by using Brave. I wanted a browser that didn't opt me into random things - period. I want a "no bullshit" web experience. Give me an address bar and a means to enable/disable things websites do, including ads, mic/cam, etc.

That's all I want. It seems to be impossible to ask anyone to do that these days, including the Brave team it seems.

**Comment by maikeldotuk** [comment]
I couldn't agree more. I've ended up with Firefox and Tor Browser.On 1 Jan 2019 18:55, Qix <notifications@github.com> wrote:
Anonymity is the freebie, convincing you of using the rewards is the real agenda

Then what's the point of switching to brave? I'm not joining a cause by using Brave. I wanted a browser that didn't opt me into random things - period. I want a "no bullshit" web experience. Give me an address bar and a means to enable/disable things websites do, including ads, mic/cam, etc.
That's all I want. It seems to be impossible to ask anyone to do that these days, including the Brave team it seems.

**Comment by TontyTon** [comment]
Hey I recently found this - "brave-browser --disable-brave-rewards-extension", used this and it worked. Brave rewards icon went away and Task manager also doesn't show it. I am on release build. Why nobody is mentioning it?
[https://github.com/brave/brave-browser/pull/1104/](url)

**Comment by Qix-** [comment]
Thanks @TontyTon but it should also be an option in the UI somewhere.

**Comment by srirambv** [comment]
The command line switch is for testing purpose.

**Comment by TontyTon** [comment]
testing thing added to master branch??!!

**Comment by maikeldotuk** [comment]
Because the whole idea is to get you on board of the rewards scheme. Privacy is the giveaway to lure you in.On 4 Jan 2019 09:47, Tonty <notifications@github.com> wrote:testing thing added to master branch??!!

**Comment by maikeldotuk** [comment]
For those still working this out, it is true using the latest dev you can run it without the rewards (FINALLY). So go to the [download page for the dev version here](https://brave-browser.readthedocs.io/en/latest/installing-brave.html#development-channel-installation) and get it. 

Then you can run it with 
brave-browser-**dev** --disable-brave-rewards-extension

**Comment by 2635599** [comment]
and if we just want to update to the latest dev without uninstalling? thanks

**Comment by TontyTon** [comment]
@maikeldotuk Hey, I meant that "--disable-brave-rewards-extension" is working in release version and **I am using it**.

**Comment by maikeldotuk** [comment]
For some strange reason it didn't work on my Ubuntu 18.01 on release version, I only managed to get it working on dev version.Either way, a second trouble got me using Firefox and TOR with two profiles , default and toruser. The later uses the default proxy of TOR and hence I can access .onion sites with it. Then mapped two keyboard shortcuts.Win+1 to Firefox -P defaultWin+2 to Firefox -P toruserThis allows me to save .onion addresses in the second profile. Something I couldn't do with tor browser and since I can play Netflix on Linux with Brave...I see no more point in trying to square the circle.On 5 Jan 2019 10:44, Tonty <notifications@github.com> wrote:@maikeldotuk Hey, I meant that "--disable-brave-rewards-extension" is working in release version and I am using it.

**Comment by headjoe3** [comment]
I almost switched to brave today.

Remove this and I will. If not, I'll just stay on Opera.

My choice of browser is literally an aesthetic choice. I want minimalism. No bells and whistles. Nothing flashy or distracting. Just pure productivity.

**Comment by TontyTon** [comment]
@headjoe3 see this, knowingly working on windows release version. Just add '-disable-brave-rewards-extension' at the end of 'Target' in the brave browser shortcut properties.

https://github.com/brave/brave-browser/issues/1475#issuecomment-451389219

**Comment by 2635599** [comment]
i'm with everyone else and would like to be able to completely disable and hide adblock and rewards. i've noticed that brave likes to block some sites, but if i turn off my vpn the site loads fine. i personally believe that something in your adblock is causing the issue, cause in chrome i do not have to disable my vpn. as for rewards regardless of claims i'll never be comfortable with ads, which is why i have adblock plus and ublock origin installed. i also noticed that on some sites you cannot set the master switch to "shield down for this site". removing EXTREMELY useful items like the full sync that is part of chromium, and the great translator that not only translates the page you are on but EVERY page on that site to add CRAPWARE to the browser is wrong. might want to take a look at cent browser for windows and see how things are actually done. tontyton having to edit the shgortcut to make changes on our end isn't the answer.

**Comment by TontyTon** [comment]
what is the problem in editing the shortcut? You can add as many things as you want.
You can disable adblocker from settings, whats the problem.
If you want to remove all shields completely, then what is the difference between Brave browser and chrome, still you can do that, again command line switch- "--disable-brave-extension"

[Personal view - Brave browser is a privacy browser, so translator needs to be removed. (rewards is an exception here, I know but they (contributors) added the commad line switch and working on adding it in settings]

As of adblock issue, you should create an 'issue'.

**Comment by 2635599** [comment]
really? just like chrome with the google extensions they have preinstalled should reside in the extensions page and be easily toggled on and off from there. having to edit the shortcut every time you want to make a change to the browser IS asinine.

**Comment by TontyTon** [comment]
Yes, I don't understand why they added these extensions that way. They should have added them as normal extensions and increased the priority (to work on webstore, only priority I detected, or remove the limitations of all extensions, they were put on by Google for its interests).

**Comment by NejcZdovc** [comment]
+1 from https://github.com/brave/brave-browser/issues/2966

**Comment by Qix-** [comment]
For anyone looking for what I described in https://github.com/brave/brave-browser/issues/1475#issuecomment-450749892, check out https://github.com/Eloston/ungoogled-chromium. There's a bit of overhead to get extensions installed, for example, but it's pretty no-nonsense if you don't mind having that bit of one-time overhead.

Been using it for a week or so now and liking it.

**Comment by 2635599** [comment]
braves problems have nothing to do with google. braves problems start with adblocker with no master off switch, a rewards program that's forced on you with no master off switch, webtorrents added with no master off switch, something under the hood just different enough for every other chromium based browser to cause some of us to have issues playing videos on certain sites, @Qix- most peeps aren't going to want to compile and run their browser, they are going to want it ready to install right out of the gate.

**Comment by Qix-** [comment]
> most peeps aren't going to want to compile and run their browser

? They have binaries available for release, e.g. on Homebrew cask. Ungoogle'd Chromium has none of that stuff. You opt-in to ad-blocking by using something like ublock origin. Sufficient enough for me, at least.

**Comment by kolnogorov** [comment]
+1 for disable, so annoying

**Comment by chrislozeau** [comment]
+1

**Comment by iZero7** [comment]
yea its annoying af i'd still use chrome, just remove it, or just release it on chrome extensions, so user who need it can install it.

**Comment by TCB13** [comment]
+1 on this.

**Comment by marcorei** [comment]
+1 to make this extension optional

**Comment by iZero7** [comment]
i don't really understand why this bloat is being added to brave, no one asked for it , why is it being forced down the users throat.

**Comment by sobolevn** [comment]
@iZero7 because of money.

**Comment by NejcZdovc** [comment]
@Qix- if you don't enable rewards we will not save any data. Only when user opt in then we start saving data. You can also just opt in and disable auto contribute, so again no data is collected and you just do manual tips.

**Comment by TontyTon** [comment]
I disabled brave rewards after creating the wallet in one profile[call it profile A], (disabled from chrome://rewards)
![br](https://user-images.githubusercontent.com/35778430/52211206-31b76080-28af-11e9-8428-3b06dde2c152.PNG)

and in other didn't created a wallet[call it profile B].
Both profiles A & B, are exactly same, no extensions, same settings, etc

- When I use profile A, brave uses more memory as compared to profile B. (from task manager, added memory of all process)

- Another thing I observe that brave with profile A tries to connect to some ips, which it connects only after creating wallet and disabling doesn't affect it.

With profile B since wallet is not yet created it doesn't connect to them.

**Comment by gabrielhug** [comment]
This issue's description and title refer only to hiding the rewards icon cosmetically. However, several merged issues referred to disabling the built-in Brave Rewards extension entirely when not in use, similar to [#1104](https://github.com/brave/brave-browser/issues/1104). See merged issues [#2789](https://github.com/brave/brave-browser/issues/2789) and [#2966](https://github.com/brave/brave-browser/issues/2966).

If hiding the rewards icon will also completely disable Brave Rewards then this issue should not have been renamed. If this solution will be purely cosmetic then the issues should likely be separated again.

**Comment by tomlowenthal** [comment]
When Rewards isn't enabled, it doesn't do anything. It's disabled! Again: Brave Rewards is _already_ disabled when not in use. Rewards is only enabled when you affirmatively click a button or flip a switch. You have to turn it on before it does anything.

Right now, the icon is visible whether or not Rewards is enabled. This issue is about hiding the icon when rewards isn't enabled. This is a cosmetic issue — hiding the icon. It doesn't change Brave Rewards' fundamental behavior, which is that Rewards doesn't do anything unless you deliberately enable it.

There isn't any need for another issue about disabling Brave Rewards, because that's already how Brave Rewards works. It's fully disabled unless you decide to turn it on.

**Comment by gabrielhug** [comment]
The browser Task Manager (Shift-Esc) shows the Brave Rewards extension maintaining a separate Process ID and memory footprint of about 20MB even when not in use. Although I understand it doesn't do anything unless configured, it is not currently possible to disable the built-in extension itself. Some (including #2789 and #2966) have asked about removing/disabling this perceived "bloat". I think these relate more directly with #1104 and should not have been merged here.

**Comment by davidtemkin** [comment]
Thanks, helpful @gabrielhug . That's an interesting piece of information. That said, much of Brave Rewards is *not* packaged as an extension; the code is in the core browser -- but should be completely inactive when Rewards isn't on.

**Comment by S0ulan** [comment]
Any news on getting rid of the icon?

**Comment by tomlowenthal** [comment]
This change to hide the icon was in https://github.com/brave/brave-core/commit/76d0644a871bd746ed33a087bc3dbb6369c5759c, which is currently in beta. If you're using beta or dev builds of Brave, you should see a setting to hide the Rewards icon.

<img width="760" alt="screen shot 2019-02-20 at 11 31 05 am" src="https://user-images.githubusercontent.com/639937/53118896-082f4380-3503-11e9-8058-a138ae49af70.png">

**Comment by btlechowski** [comment]
Handled in https://github.com/brave/brave-browser/issues/2975. The fix is part of current Beta (0.61.x).

**Comment by beyera** [comment]
Thanks! Just started using Brave today and was about to uninstall until I found out this was in the beta. ❤️

**Comment by Rundik** [comment]
If I understood it right they say you can't remove this thing at all, you can only hide it. Even in chrome you can decide which extensions to have, but not in brave

**Comment by bsclifton** [comment]
@Rundik that's correct for the time being - we do have a task captured to terminate the process when disabled:
https://github.com/brave/brave-browser/issues/3436

**Comment by iZero7** [comment]
just fcking delete it, no one asked for it

**Comment by iZero7** [comment]
anyone can guide me on how to build brave from source without brave rewards??

**Comment by iZero7** [comment]
thanks for deleting my comment, i get it, bye forever

**Comment by Rundik** [comment]
The idea to build brave from source without brave rewards is kind of interesting

**Comment by gabrielhug** [comment]
The issue created for disabling the rewards extension entirely rather than just hiding the icon is: #3436 
I'd suggest at least giving it a thumbs-up if it matters to you. It'd likely be helpful to carry on related conversation there rather than here on this closed issue.

**Comment by simonhong** [comment]
@iZero7 
> anyone knows how to build brave from source without brave rewards?

I think brave rewards would be disabled by modifying `brave/components/brave_rewards/browser/buildflags/buildflags.gni` like below.

`brave_rewards_enabled = false`

**Comment by pvlg** [comment]
Another attempt to start using brave browser failed

**Comment by martynchamberlin** [comment]
I'm confused here. I'm on Brave 0.66.99 and I am not seeing a place where I can hide shields. This issue makes me think I should be seeing that as a setting. Any ideas?

**Comment by bsclifton** [comment]
@martynchamberlin there is no way to hide the shields icon - you can hide the BAT icon (the triangle) though

**Comment by martynchamberlin** [comment]
@bsclifton Gotcha, thanks.  If I had my way, I'd have the ability to remove that badge as well as the bookmark icon on the left of the address bar. And have a way to disable the auto-hiding of the `https://www.` part of the domain...

**Comment by bsclifton** [comment]
@martynchamberlin disabling the hiding of `https://www` is on the way (already fixed in Dev and Nightly channels).

**Comment by mikhoul** [comment]
> @martynchamberlin disabling the hiding of `https://www` is on the way (already fixed in Dev and Nightly channels).

I'm new with Brave, where will be this setting in the release channel ?

**Comment by rebron** [comment]
@mikhoul Won't be a setting, we'll be showing full url starting next week with 68.x.
https://github.com/brave/brave-browser/issues/2192

**Comment by HenrySqwidbern** [comment]
Can I bump this if it's closed?

What about a control to hide the icon, whether rewards/wallet function is enabled or not.

It would be nice to be able to use at school work wherever without explaining to any over-the-shoulder-lurker-berzerkers who might have a policy-paranoid opinion to express. 

I wish to use Brave on every computing device that my hands touch, no compromises.

**Comment by 2635599** [comment]
even those that don't belong to you. that vandalizing property that doesn't belong to you.

**Comment by HenrySqwidbern** [comment]
> even those that don't belong to you. that vandalizing property that doesn't belong to you.

I think you mean:
Even those that don't belong to you? That is vandalizing property that doesn't belong to you.

Yes.
But cast my evil use case aside.
I suspect an example use case involving a person living under an oppressive regime and supporting the same ask could be constructed that would comply with your stringent morals.

**Comment by 2635599** [comment]
don't play grammar nazi IT'S RUDE, jast as is changing the settings on others computers just because you don't like them.

**Comment by nickenchev** [comment]
In a highly competitive spaces (web browsers), it makes no sense to me when developers put in "features" like this. People are using your browser because they don't want bullshit ads or any kind of dumb bloat, or extra "crap" that they have to decline, hide, or close. When you include some dumb extension that most people don't want, and you make it impossible or purposely difficult to hide, you kill your brand. 

"It takes 20 years to build a reputation and five minutes to ruin it. If you think about that, you’ll do things differently."
- Warren Buffett

**Comment by pvlg** [comment]
For this reason, I still use chrome.

**Comment by Rundik** [comment]
me too

**Comment by bsclifton** [comment]
@nickenchev @pvlg @Rundik maybe I'm missing something - but you can actually hide the BAT icon. We implemented it with another issue, but basically you can find it under brave://settings
![image](https://user-images.githubusercontent.com/4733304/90219788-ac720d80-ddbb-11ea-85b7-e383d873caf3.png)

As long as you don't enable rewards, you can hide the icon. It also shouldn't load up if you check the browser's task manager (Hamburger menu => More tools => Task manager)

**Comment by oliverbenns** [comment]
@bsclifton Why do we have to search to disable this? As others have mentioned no-one wants Binance, no-one wants Brave rewards. Why would a user honestly go through leaps and bounds to get a standard browser without ads literally built in that won't go from the address bar? Next you'll be adding an Ask Jeeves toolbar.

**Comment by Willardgmoore** [comment]
Hide Brave Rewards button should be turned on by default. It pissed me off enough that I would not ever consider brave if it weren't for someone else on my dev team using it. I am not looking to remove the shields up icon. I don't care for bubbles with icons. I want a browser where I can be here to work and not be bothered. This has been a really bad experience for a new user. and I am not excited about figuring out this freaking icon and it's tiny number as well.

<img width="52" alt="Screen Shot 2021-04-09 at 4 58 56 PM" src="https://user-images.githubusercontent.com/7650197/114248438-e3052580-9954-11eb-8719-b7376ef0e6e7.png">

**Comment by Redsandro** [comment]
In my opinion the Brave Rewards button can be visible by default in order to give the Brave devs a chance to onboard some people, but upon clicking on it there should really be an option to opt-out completely, hiding the Brave Rewards button.

**Comment by Robanan** [comment]
So there is no way to disable this right? my only concern is that it's taking cpu power, atm I'm trying to save on cpu power as much as I can.

**Comment by amacgillivray** [comment]
> So there is no way to disable this right? my only concern is that it's taking cpu power, atm I'm trying to save on cpu power as much as I can.

See https://github.com/brave/brave-browser/issues/1475#issuecomment-673909395

Anyways, having something like Brave Rewards visible on the URL bar makes me second guess whether or not the browser is appropriate for work. Seemed iffy. The method to hide it is still not discoverable enough; showing the option to disable it on the brave://rewards page would really help people discover that menu item.

**Comment by j-rodman** [comment]
Right clicking on the rewards button in the url bar should lead to a means of hiding it directly.

**Comment by 174n** [comment]
The issue isn't about annoying icon, but about an extension that can't be removed

**Comment by jcmatoskx** [comment]
Took a while to find where to disable, as it's not on brave://settings but on  brave://settings/appearance
The weirdest thing is why are these icons in such bright colours, distracting from everything else in the page. If Brave insists on having them, why aren't they less prominent. If they would be in gray I probably wouldn't have even bothered to search how to disable them and would have learned to live with them, maybe even accept their features in the future. 
But it's my first time trying the browser and it just looks and feels like I have some preinstalled bloat. 
If I'm reading a plain black and white page, I have two extremely saturated colourful playful icons on top, makes the browser look like for kids, or like I have some malware games installed. 

Also, even though Brave rewards are disabled, they still very clearly show in the task manager as running
![image](https://user-images.githubusercontent.com/37616081/149295761-2eb6c64f-2a21-4ba6-87fb-654f12e2de7d.png)

**Comment by iefremov** [comment]
@jcmatoskx have you restarted the browser after disabling rewards? The extension shouldn't be loaded after restart - could you please recheck?

**Comment by MaybeAnonymous** [comment]
I agree. I will not use Brave until I have the choice to disable that. Freedom of software is important

**Comment by jcmatoskx** [comment]
> @jcmatoskx have you restarted the browser after disabling rewards? The extension shouldn't be loaded after restart - could you please recheck?

Before posting, for confirming, I purposely had restarted the browser several times but the task had remained open. Currently It appears to not be running anymore :)

**Comment by iefremov** [comment]
@MaybeAnonymous there is a choice, please make yours
![image](https://user-images.githubusercontent.com/3109556/151759660-d9a9bc96-86f7-4487-952c-c1997043a098.png)

**Comment by MaybeAnonymous** [comment]
> @MaybeAnonymous there is a choice, please make yours
> ![image](https://user-images.githubusercontent.com/3109556/151759660-d9a9bc96-86f7-4487-952c-c1997043a098.png)
> 

i mean the brave rewards process still runs

**Comment by iefremov** [comment]
> i mean the brave rewards process still runs

it shouldn't run in the modern versions. Could you please recheck @MaybeAnonymous ?

**Comment by sdrinf** [comment]
Just leaving a note here.
Today, I gave brave a try.
It fucking attention-stealing by having a red glowing lion icon in the right side of the bar, all the time.
Moving down to ungoogled-chromium next.
GL with your video game.
